flang/lib/Optimizer/CodeGen/BoxedProcedure.cpp

```diff
@@ -170,9 +181,10 @@ class BoxedProcedurePass : public BoxedProcedurePassBase<BoxedProcedurePass> {
     if (options.useThunks) {
       auto *context = &getContext();
       mlir::IRRewriter rewriter(context);
-      BoxprocTypeRewriter typeConverter;
+      BoxprocTypeRewriter typeConverter(mlir::UnknownLoc::get(context));
       mlir::Dialect *firDialect = context->getLoadedDialect("fir");
       getModule().walk([&](mlir::Operation *op) {
+        typeConverter.setLocation(op->getLoc());
```

Comment:
Yeah, while this looks kind of hacky, given the structure of `mlir::TypeConverter::convertType` it seems like a workaround to pass the location information.